### DISCUSSION TITLE
Is 5.1.0 version can work with older versions of android and react native?

### DISCUSSION THREAD

**Discussion by elad30300** [react-native-image-picker/react-native-image-picker#2086] [Q&A]
Hi,
I have an app that uses react native v0.63.3.
I tried to use the latest version (5.1.0) to open the gallery with my app and the device I used is Samsung S10 with Android 10. When I called `launchImageLibrary` I got: `cannot read property 'launchImageLibrary' of null`, and after further debugging I found out that `NativeModule.ImagePicker` is `null`.
When I used version 4.4.2 it worked and let me use the images gallery.

I would like to know the following things:
1. Is 5.1.0 version works with react native v0.63.3 or it is too old? If it is too old, what is the highest version that should work?
2. My app is for devices with android version 10, 11, 12 (in future maybe 13) - what is the highest version that should work on all of them?

Thanks ahead!

**Comment by Johan-dutoit** [comment]
👋  I don't see why it shouldn't work with 63.3, although so much has changed since then and my memory is likely forgetting something!

Assuming you've done the usual `npx pod-install ios` (or `pod install`) after the upgrade?  Have you tried creating a new project with v0.63.3 and only installing image picker?

**Reply by elad30300** [reply]
Hi,
First of all, thanks for the fast response!
Actually my app is only for android so i didnt do the pod install.

I haven't tried creating new project and install only this library, but why does it matter?

**Reply by Johan-dutoit** [reply]
Ah, okay, no worries about pod install... that's usually the first problem people experience with these kind of issues.
The reason for creating a new project is to see if the problem is related to your specific project or the library itself, or if it's compatibility related.

**Reply by elad30300** [reply]
Hi, it still does the same thing even with empty project. I think it is because in version 5.1.0 the assumption is there is no need for linking because of auto linking, so even if I do `react-native link` it still doesn't link the library in the android files, but maybe my react native version is too old that I actually need linking. For example, in version 4.4.2 there is linking and it works.

**Reply by elad30300** [reply]
@Johan-dutoit Maybe can you tell what is the closest version to the latest that should work under my circumstances? 🙏
